spec/integration/federation/shared_receive_retraction.rb

```diff
@@ -1,43 +1,62 @@
-shared_examples_for "it retracts non-relayable object" do
-  it "retracts object by a correct retraction message" do
-    target_klass = target_object.class.to_s.constantize
-    Workers::ReceiveEncryptedSalmon.new.perform(@user.id, generate_retraction(entity_name, target_object))
-
-    expect(target_klass.exists?(guid: target_object.guid)).to be(false)
+def mock_private_keys_for_retraction(entity_name, entity, sender)
+  if %i(signed_retraction_entity relayable_retraction_entity).include?(entity_name)
+    allow(DiasporaFederation.callbacks).to receive(:trigger)
+                                             .with(:fetch_private_key_by_diaspora_id, sender.diaspora_handle)
+                                             .and_return(sender.encryption_key)
   end
+  if entity_name == :relayable_retraction_entity
```

Comment:
Use a guard clause instead of wrapping the code inside a conditional expression.